File: qr-auth-service/src/main/java/com/app/authentiScan/payload/response/QrVerificationHistoryResponse.java
package com.app.authentiScan.payload.response;

import com.app.authentiScan.enums.VerificationStatusEnum;
import lombok.*;
import java.time.LocalDateTime;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class QrVerificationHistoryResponse {
    private Long id;
    private String uid;
    private Integer sl;
    private VerificationStatusEnum verificationScanStatusEnum;
    private Integer qrVerificationCountScan;
    private LocalDateTime lastQrVerificationScanAt;
    private String lastQrVerificationScanAtFormatted;
    private VerificationStatusEnum verificationSmsStatusEnum;
    private Integer qrVerificationCountSms;
    private LocalDateTime lastQrVerificationSmsAt;
    private String lastQrVerificationSmsAtFormatted;
    private LocalDateTime updatedAt;
    private String brandKey;
}

File: qr-auth-frontend/src/app/_model/response/qr-verification-history-response.ts
import {VerificationStatusEnum} from '../enums/verification-status-enum';

export class QrVerificationHistoryResponse {
  id!: number;
  uid!: string;
  sl!: number;
  verificationScanStatusEnum!: VerificationStatusEnum;
  qrVerificationCountScan!: number;
  verificationSmsStatusEnum!: VerificationStatusEnum;
  qrVerificationCountSms!: number;
  updatedAt!: string;
  lastQrVerificationScanAt!: string;
  lastQrVerificationScanAtFormatted!: string;
  lastQrVerificationSmsAt!: string;
  lastQrVerificationSmsAtFormatted!: string;
  brandKey!: string;
  isLoading!: boolean;

  public constructor(o?: Partial<QrVerificationHistoryResponse>) {
    Object.assign(this, o);
  }
}
